### ISSUE TITLE
Modern toolbar UI is not working after saving window configuration on window with Shrink mode

### ISSUE THREAD

**Issue by gileli121** [gileli121/WindowTop#48]
A user reported that the new toolbar UI is not showing even when enabling the modern toolbar feature.

I asked the user for the WindowTop.settings file, and he sent it.
I found that the configuration that saved in this file was reproduced the issue.

While debugging the issue, I found that the configuration key that reproduces it is:
```
[WindowsSettings]
=ArrowXPos=0.45
```

It should not be like that. it should be:
```
[WindowsSettings]
some_process.exe=ArrowXPos=0.45
```

I reviewed the code, and according to the code review, this configuration issue is caused in the following scenario:

1. Disable the modern toolbar
2. Shrink some window
3. Resize the shrink box so the arrow button will show when you move to the top area...
4. Drag the arrow button to some another position
5. Unshrink the window or exit WindowTop

At this point, this configuration issue will reproduce.
Because this configuration issue affects only the modern toolbar UI, you will need to enable the modern toolbar UI feature to reproduce it.

Seems that the user did this scenario in older version or he did it before enabaling the modern toolbar feature.

**Comment by gileli121** [comment]
It will now save it like this:
```
[WindowsSettings]
WT_SHRINKBOX=ArrowXPos=0.45
```

**Comment by sflorg** [comment]
Though I don't recall having any config saved in a shrink mode. The only saved window is with "Sage" in dark mode. Anyhow you are quick to work on issues. Well done. 
Heidi-Ann

**Comment by gileli121** [comment]
It can happen by saving window config or by dragging the arrow button (on the "shrink box" and not on the window itself)

In the C# side (not Autoit), it will change to

```c#
public WindowItem(IntPtr hwnd, bool isShrinkBox = false)
{
	this.Hwnd = hwnd;

	if (!isShrinkBox)
	{
		Process process = Win32Wrapper.GetProcessByHwnd(hwnd);
		this.ProcessName = process.ProcessName + ".exe";
	}
	else
	{
		this.ProcessName = "WT_SHRINKBOX";
	}

	Settings = Logic.Settings.Windows.GetWindowSettings(ProcessName);

	UpdatePos();
}
```
instead of

```c#
public WindowItem(IntPtr hwnd, bool isShrinkBox = false)
{
	this.Hwnd = hwnd;
	Process process = Win32Wrapper.GetProcessByHwnd(hwnd);
	this.ProcessName = process.ProcessName + ".exe";
	Settings = Logic.Settings.Windows.GetWindowSettings(ProcessName);
	UpdatePos();
}
```
```

**Comment by gileli121** [comment]
@sflorg 
Please download v4.0.1 from here:
https://github.com/gileli121/WindowTop/releases/tag/v4.0

Let me know if it works fine.


### v4.0.1
* Fixed: Modern toolbar UI is not working after saving window configuration on a window with Shrink mode (Issue #48)
* Fixed: Issue that the modern toolbar UI may show when it should not
* Fixed: Issue that the option to disable the toolbar menu is not honored by the modern toolbar menu
* Fixed: blacklist/whitelist windows options was not honored by the modern toolbar menu
* Fixed: Removed a wrong message that in the current version you can't customize the modern toolbar menu (The message was shown in the toolbar options window)

**Comment by sflorg** [comment]
Works perfect, haven't noticed any issues other then a few programs it doesn't work with (and never did), yet that is expected in my opinion. 
Thanks for your quick attention.
Heidi-Ann

**Comment by gileli121** [comment]
@sflorg 
The fix for this bug caused a regression bug that when you enable shrink mode, the shrinking window (the "shrink box") still has the option to shrink it again... you can shrink the shrink box. It fixed now in v4.0.3

Please redownload. 
If there is any issue, let me know.

**Comment by gileli121** [comment]
Fix in v4.0.3 did another regression :( 
You should download v4.1.0 instead

**Comment by sflorg** [comment]
Somehow I don't see what happened to 4.0.2 which is what I have installed last. Is your recommendation to remove 4.0.2 and go back to 4.0.1.
